src/Container/AbstractProjectionManagerFactory.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * This file is part of the prooph/pdo-event-store.
+ * (c) 2016-2017 prooph software GmbH <contact@prooph.de>
+ * (c) 2016-2017 Sascha-Oliver Prolic <saschaprolic@googlemail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Prooph\EventStore\Pdo\Container;
+
+use Interop\Config\ConfigurationTrait;
+use Interop\Config\ProvidesDefaultOptions;
+use Interop\Config\RequiresConfig;
+use Interop\Config\RequiresConfigId;
+use Interop\Config\RequiresMandatoryOptions;
+use Prooph\EventStore\EventStore;
+use Prooph\EventStore\Pdo\Exception\InvalidArgumentException;
+use Prooph\EventStore\Projection\ProjectionManager;
+use Psr\Container\ContainerInterface;
+
+abstract class AbstractProjectionManagerFactory implements
+    ProvidesDefaultOptions,
+    RequiresConfig,
+    RequiresConfigId,
+    RequiresMandatoryOptions
+{
+    use ConfigurationTrait;
+
+    private $configId;
+
+    /**
+     * Creates a new instance from a specified config, specifically meant to be used as static factory.
+     *
+     * In case you want to use another config key than provided by the factories, you can add the following factory to
+     * your config:
+     *
+     * <code>
+     * <?php
+     * return [
+     *     ProjectionManager::class => [MySqlProjectionManagerFactory::class, 'service_name'],
+     *     // or
+     *     ProjectionManager::class => [PostgresProjectionManagerFactory::class, 'service_name'],
+     * ];
+     * </code>
+     *
+     * @throws InvalidArgumentException
+     */
+    public static function __callStatic(string $name, array $arguments): ProjectionManager
+    {
+        if (! isset($arguments[0]) || ! $arguments[0] instanceof ContainerInterface) {
+            throw new InvalidArgumentException(
+                sprintf('The first argument must be of type %s', ContainerInterface::class)
+            );
+        }
+
+        return (new static($name))->__invoke($arguments[0]);
+    }
+
+    public function __construct(string $configId = 'default')
+    {
+        $this->configId = $configId;
+    }
+
+    public function __invoke(ContainerInterface $container): ProjectionManager
+    {
+        $config = $container->get('config');
+        $config = $this->options($config, $this->configId);
+
+        $projectionManagerClassName = $this->projectionManagerClassName();
+
+        return new $projectionManagerClassName(
+            $container->get(EventStore::class),
+            $container->get($config['connection']),
```

Comment:
The `EventStore` service should also be configurable or? Default can be `EventStore::class`.